support/map.c

```diff
@@ -5237,7 +5237,7 @@ Mono_Posix_ToSockaddrIn (struct sockaddr_in *from, struct Mono_Posix_SockaddrIn
 #endif /* ndef HAVE_STRUCT_SOCKADDR_IN */
 
 
-#ifdef HAVE_STRUCT_SOCKADDR_IN6
+#if defined(HAVE_STRUCT_SOCKADDR_IN6) && !defined(HOST_WIN32)
```

Comment:
I think #4754 is just the wrong approach. The only reason why it doesn't break there is because then HAVE_STRUCT_SOCKADDR_IN6 is not defined here.